tests/unit/generate_detect_target_contour.py

```diff
@@ -0,0 +1,165 @@
+        blur: Indicates whether the landing pad should have a blur effect. default: False.
+        angle (int): The rotation angle of the landing pad in degrees. defaults: 0.
+    """
+
+    def __init__(
+        self,
+        center: tuple[int, int],
+        axis: tuple[int, int],
+        blur: bool = False,
+        angle: int = 0,
+    ):
+
+        self.center = center
+        self.axis = axis
+        self.blur = blur
+        self.angle = angle
+
+
+def blur_image(background: np.ndarray, landing_data: LandingPadData) -> NumpyImage:
+    """
+    Blurs an image a singular shape, adds it to the background, and returns an image.
+    """
+
+    background_copy = background.copy()
+    x, y = background_copy.shape[:2]
+
+    mask = np.zeros((x, y), np.uint8)
+    mask = cv2.ellipse(
+        mask,
+        landing_data.center,
+        landing_data.axis,
+        landing_data.angle,
+        0,
+        360,
+        255,
+        -1,
+    )
+
+    mask = cv2.blur(mask, (25, 25), 7)
+
+    alpha = mask[:, :, np.newaxis] / 255.0
+    # Brings the image back to its original color
+    fg = np.full(background.shape, LANDING_PAD_COLOR, dtype=np.uint8)
+
+    blended = (background * (1 - alpha) + fg * alpha).astype(np.uint8)
+    return NumpyImage(blended)
+
+
+def draw_landing_pad(
+    image: np.ndarray, landing_data: LandingPadData
+) -> tuple[NumpyImage, BoundingBox]:
+    """
+    Draws an ellipse on the provided image and saves the bounding box coordinates to a text file.
+    """
+    (h, k), (a, b) = landing_data.center, landing_data.axis
+    rad = math.pi / 180
+    ux, uy = a * math.cos(landing_data.angle * rad), a * math.sin(landing_data.angle * rad)
+    vx, vy = b * math.sin(landing_data.angle * rad), b * math.cos(landing_data.angle * rad)
+    width, height = 2 * math.sqrt(ux**2 + vx**2), 2 * math.sqrt(uy**2 + vy**2)
+
+    top_left = (int(max(h - (0.5) * width, 0)), int(max(k - (0.5) * height, 0)))
+    bottom_right = (
+        int(min(h + (0.5) * width, image.shape[1])),
+        int(min(k + (0.5) * height, image.shape[0])),
+    )
+
+    if landing_data.blur:
+        image = blur_image(image, landing_data)
+        return image, BoundingBox(top_left, bottom_right)
+
+    image = cv2.ellipse(
+        image,
+        landing_data.center,
+        landing_data.axis,
+        landing_data.angle,
+        0,
+        360,
+        LANDING_PAD_COLOR,
+        -1,
+    )
+    return NumpyImage(image), BoundingBox(top_left, bottom_right)
+
+
+def create_test(landing_list: list[LandingPadData]) -> LandingPadTestData:
+    """
+    Generates test cases given a data set.
+    """
+    image = np.full(shape=(1000, 2000, 3), fill_value=255, dtype=np.int16)
+
+    boxes_list = []
+    for landing_data in landing_list:
+        np_image, bounding_box = draw_landing_pad(image, landing_data)
+        image = np_image.image
+
+        boxes_list.append([1, 0] + list(bounding_box.top_left + bounding_box.bottom_right))
```

Comment:
What is the purpose of `[1, 0]` ? Make it so that the argument to `append()` is a variable as well.